MariaDB/server
# MDEV-35599: Replace (`u`)`llstr()` with `%llu`/`d`

* [x] *The Jira issue number for this PR is: [MDEV-35599](https://jira.mariadb.org/browse/MDEV-35599)*

## Description
> `llstr()` is a workaround for printing longlongs from the times when `%llu` was not supported
> ⸺ @vuvova, https://github.com/MariaDB/server/pull/3360#discussion_r1833301827

### What problem is the patch trying to solve?
Nearly all of (`u`)`llstr()` calls were preparations for calls to the `printf` extended family; but both the C/C++ `printf` family and our `my_snprintf()` & descendants already support `long long`s through the `ll` size specifier.
Inlining `llstr()` saves an additional function call each *everywhere*, which
* reduces cognitive load
* potentially improves performance
* mitigates buffer overruns from *if*[^if] the `long long` is too large for the buffer given to `llstr()`

#### Real motivation
I wanted to modernize our ancient code.[^fun]
In addition, I visited lots of files across components during project-wide cleanups like this and #3360, which helps me familiarize with the repository (and its state of maintenance) as a new recruit of the Corp.

#### Non-goals
* Check the correctness of the types with or without this rewrite
  * Particularly, I feel that many `llstr()` calls (replaced with `"%lld"`) should rather use `ullstr()` (which would be `"%llu"`)
    * There’s only a few `ullstr` calls currently.
    * Turning `-Wformat-signedness` on will warn[^Werror] these mistakes.
    * `"%"PRId64`, `"%"PRIu64` and `%zu` are likely more accurate.
  * Many also explicitly casted their values to `longlong`
  * I’ve only corrected cases that emitted `-Wformat`[^Wformat] complaints, such as `sql/sql_test.cc`
    * Some latter files used `llstr` to convert `int`s and `long`s to strings… ¯\\\_(ツ)\_/¯
* Extensive reformat our ancient files
  * You can tell a file’s gone through a history from its mix of tab-space and space-only indents.
    * Speaking of indents – I used a mix of preferences in my diffs… I should chat with others on our preferences vs. practicality.

## Release Notes
TODO: *Ask QA engineers for their impressions on this housekeeping*

## How can this PR be tested?
* Manual: Check the deletion of each `llstr()` usage that it’s either
  * inlined into its outer `printf` by replacing `%s` with `%lld`
  * or, seldomly (a dozen out of a few hundred), entirely replaced with `snprintf(…, "%lld", …)`
* Automatic:
  1. Optional: Create to a temporary branch to merge #3360 with this commit[^Wformat]
  2. Build the server and verify there’re no compile warnings[^Werror]
  3. Run the tests to verify that no output or behavior were affected

## Basing the PR against the correct MariaDB version
* [x] *This is a new feature or __a refactoring__, and the PR is based against the `main` branch.*

## PR quality check

[^if]: I don’t think we have one right now… :face_in_clouds:
[^fun]: See also: https://justforfunnoreally.dev
[^Werror]: We also have `-Werror` active.
[^Wformat]: `-Wformat` currently only supports the C/C++ `printf` family; #3360 of [MDEV-21978](https://jira.mariadb.org/browse/MDEV-21978) will add support for `my_vsnprintf` and descendants.
